sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/InsertIntoHadoopFsRelationCommand.scala

```diff
@@ -148,6 +148,9 @@ case class InsertIntoHadoopFsRelationCommand(
         options = options)
 
       fileIndex.foreach(_.refresh())
+      catalogTable.foreach { table =>
+        sparkSession.sharedState.cacheManager.uncacheQuery(sparkSession.table(table.identifier))
```

Comment:
This PR can make the behavior consistent with what we did for [insertion of Hive serve tables](//github.com/apache/spark/blob/master/sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/InsertIntoHiveTable.scala#L378)